src/physical_object.py

```diff
@@ -45,24 +45,80 @@ def get_forces_composition(self):
             force[1] += self.forces[f][1]
         return force
 
+    def reset_forces(self):
+        for f in self.forces:
+            self.forces[f] = [0., 0.]
+
+    def add_force(self, label, force):
+        if label not in self.forces:
+            self.forces[label] = [0., 0.]
+        f = self.forces[label]
+        f[0] += force[0]
+        f[1] += force[1]
+
+    @staticmethod
+    def distance(p1, p2):
+        """
+        Calculate the distance between two objects.
+
+        :p1: first object
+        :p2: second object
+
+        :returns: distance
+        """
+        return sqrt((p1.x - p2.x)**2 + (p1.y - p2.y)**2)
+
     @staticmethod
-    def collide(p, q):
+    def calc_gravity(p1, p2):
+        """
+        Calculate the gravity force vector between two objects.
+
+        :p1: first object
+        :p2: second object
+        """
+        static = PhysicalObject
+        d = static.distance(p1, p2)
+        m1, m2 = p1.mass, p2.mass
+        r1, r2 = p1.radius, p2.radius
+
+        def _intersection_coefficient():
+            if d >= r1 + r2:
+                return [1., 1.]
+            r = r1 + r2 - d
+            v = r*r*(PI/12.)*(d*d + 2*d*(r1+r2) - 3*((r1-r2)**2))/d
```

Comment:
V is volume, R is the "length" of intersection (aka r1=radius1 + r2=radius2 - d=distance)